include/swift/Driver/Driver.h

```diff
@@ -247,20 +249,22 @@ class Driver {
 
   /// Construct the OutputFileMap for the driver from the given arguments.
   std::unique_ptr<OutputFileMap>
-  buildOutputFileMap(const llvm::opt::DerivedArgList &Args) const;
+  buildOutputFileMap(const llvm::opt::DerivedArgList &Args,
+                     StringRef workingDirectory) const;
 
   /// Add top-level Jobs to Compilation \p C for the given \p Actions and
   /// OutputInfo.
   ///
   /// \param TopLevelActions The main Actions to build Jobs for.
   /// \param OI The OutputInfo for which Jobs should be generated.
   /// \param OFM The OutputFileMap for which Jobs should be generated.
+  /// \param workingDirectory If non-empty, used to resolve any generated paths.
   /// \param TC The ToolChain to build Jobs with.
   /// \param C The Compilation containing the Actions for which Jobs should be
   /// created.
   void buildJobs(ArrayRef<const Action *> TopLevelActions, const OutputInfo &OI,
-                 const OutputFileMap *OFM, const ToolChain &TC,
-                 Compilation &C) const;
+                 const OutputFileMap *OFM, StringRef workingDirectory,
+                 const ToolChain &TC, Compilation &C) const;
```

Comment:
I think I'd prefer to make this a field on the Compilation object. It's of a similar provenance to the other things stored there and it looks like you're having to thread it through lots of calls that already take a Compilation.

Reply:
Oh fun.  I was following the example of `OutputFileMap`, `OutputInfo` and `ToolChain`.  Now that I look more carefully at `Compilation`, I see we have fields for all three of those (and they appear to me to be the same objects), so I don't know why we also thread them through every call too.  I'm happy to move working dir into Compilation assuming @jrose-apple agrees.

Reply:
I don't love it because it seems like something that's only used while building up the Compilation, and not something you need once it's fully-built and ready to, well, compile things. But maybe that distinction's only in my head.

(Originally, none of the three things you mentioned were in Compilation. They've all migrated there because of batch mode, which needs to construct jobs "late".)

Reply:
The OutputFileMap held in the compilation is a derived one (holding all the aux and intermediates in a compilation), whereas the one being passed around as args here is an input. This is a somewhat recent development and perhaps not ideal, but they're not the same object.

The ToolChain and OutputInfo args _are_ the same as those being threaded around though, it's true; I guess we can just pull them out of Compilation now (they're new arrivals).

Reply:
Oh, ok. On matters of taste here I'm happy to defer to @jrose-apple, passed-as-an-arg it is!

Reply:
I'll admit I don't love another argument being passed around either. I'm just not sure Compilation is the place to put it. And Ben doesn't have to be the one to clean this up.